src/scripts/run_certbot.sh

```diff
@@ -49,12 +49,39 @@ fi
 # $1: The name of the certificate (e.g. domain-rsa)
 # $2: String with all requested domains (e.g. -d domain.org -d www.domain.org)
 # $3: Type of key algorithm to use (rsa or ecdsa)
+# $4: The authenticator to use to solve the challenge
 get_certificate() {
-    info "Requesting an ${3^^} certificate for '${1}'"
+    local challenge_authenticator="$4"
+    local challenge_params=
+    local challenge_type=
+
+    if [ -n "$challenge_authenticator" ] && [[ "$challenge_authenticator" == dns-* ]]; then
+        local provider="${challenge_authenticator#dns-}"
+        local configfile="/etc/letsencrypt/${provider}.ini"
+        if [ -f "$configfile" ]; then
+            challenge_type="dns-01"
+            challenge_params="--${challenge_authenticator}-credentials=${configfile}"
+            local challenge_envvar="CERTBOT_DNS_${provider^^}_PROPAGATION_SECONDS"
```

Comment:
This line can be removed if you change to the more generic name suggested above.